src/main/java/org/duckdb/DuckDBResultSet.java

```diff
@@ -1286,7 +1295,19 @@ public <T> T getObject(int columnIndex, Class<T> type) throws SQLException {
     }
 
     public <T> T getObject(String columnLabel, Class<T> type) throws SQLException {
-        throw new SQLFeatureNotSupportedException("getObject");
+        if (type == null) {
+            throw new SQLException("type is null");
+        }
+        if (columnLabel == null || columnLabel.isEmpty()) {
+            throw new SQLException("columnLabel is null");
+        }
+        int index = -1;
+        for (int i = 0; i < meta.column_names.length; i++) {
```

Comment:
Please use the existing findColumn method here instead

Reply:
add some unit test in TestDuckDBJDBC and use findColumn